db/query_test.go

```diff
@@ -568,3 +569,185 @@ func countQueryResults(results sgbucket.QueryResultIterator) int {
 	}
 	return count
 }
+
+func TestQueryChannelsActiveOnlyWithLimit(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test require Couchbase Server")
+	}
+
+	db, testBucket := setupTestDB(t)
+	defer testBucket.Close()
+	defer tearDownTestDB(t, db)
+
+	docIdFlagMap := make(map[string]uint8)
+	var startSeq, endSeq uint64
+	body := Body{"channels": []string{"ABC"}}
+
+	checkFlags := func(entries LogEntries) {
+		for _, entry := range entries {
+			assert.Equal(t, docIdFlagMap[entry.DocID], entry.Flags, "Flags mismatch for doc %v", entry.DocID)
+		}
+	}
+
+	// Create 10 added documents
+	for i := 1; i <= 10; i++ {
+		id := "created" + strconv.Itoa(i)
+		doc, revId, err := db.PutExistingRevWithBody(id, body, []string{"1-a"}, false)
+		require.NoError(t, err, "Couldn't create document")
+		require.Equal(t, "1-a", revId)
+		docIdFlagMap[doc.ID] = uint8(0x0)
+		if i == 1 {
+			startSeq = doc.Sequence
+		}
+		endSeq = doc.Sequence
+	}
+
+	// Create 10 deleted documents
+	for i := 1; i <= 10; i++ {
+		id := "deleted" + strconv.Itoa(i)
+		doc, revId, err := db.PutExistingRevWithBody(id, body, []string{"1-a"}, false)
+		require.NoError(t, err, "Couldn't create document")
+		require.Equal(t, "1-a", revId)
+
+		body[BodyDeleted] = true
+		doc, revId, err = db.PutExistingRevWithBody(id, body, []string{"2-a", "1-a"}, false)
+		require.NoError(t, err, "Couldn't create document")
+		require.Equal(t, "2-a", revId, "Couldn't create tombstone revision")
+
+		docIdFlagMap[doc.ID] = uint8(0x1) // 1 = Deleted(1)
```

Comment:
Should use constants from change_log.go here and below (channels.Deleted, etc) instead of rebuilding uint8 values.

Reply:
Added changes to use constants from change_log.go instead of rebuilding uint8 values.